monitoring/uss_qualifier/resources/planning_area.py

```diff
@@ -32,23 +31,43 @@ class PlanningAreaSpecification(ImplicitDict):
     If not specified, a fake URL will be generated at runtime according to the test in which the resource is being
     used."""
 
-    volume: Volume3D
-    """3D volume of service area"""
-
-    def get_volume4d(
-        self, time_start: datetime.datetime, time_end: datetime.datetime
-    ) -> Volume4D:
-        return Volume4D(
-            volume=self.volume,
-            time_start=Time(time_start),
-            time_end=Time(time_end),
-        )
-
     def get_base_url(self, frames_above: int = 1):
         if "base_url" in self and self.base_url is not None:
             return self.base_url
         return make_fake_url(frames_above=frames_above + 1)
 
+
+class PlanningAreaResource(Resource[PlanningAreaSpecification]):
+    specification: PlanningAreaSpecification
+    _volume: VolumeResource
+
+    def __init__(
+        self,
+        specification: PlanningAreaSpecification,
+        volume: VolumeResource,
+        resource_origin: str,
```

Comment:
`resource_origin` is an infrastructure argument and should [come before](https://github.com/interuss/monitoring/blob/61247cbd76b86448637054c08018960f29bb10b1/monitoring/uss_qualifier/resources/resource.py#L24 any dependencies (like in [DSSInstanceResource](https://github.com/interuss/monitoring/blob/61247cbd76b86448637054c08018960f29bb10b1/monitoring/uss_qualifier/resources/astm/f3548/v21/dss.py#L698)).

```suggestion
        specification: PlanningAreaSpecification,
        resource_origin: str,
        volume: VolumeResource,
```